### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by TsvetanKichuk**

### CODE REVIEW

File: cinema/views.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request, pk):
+    movie = get_object_or_404(Movie, pk=pk)
+    if request.method == 'GET':
+        serializer = MovieSerializer(movie)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    elif request.method == 'PUT':
+        serializer = MovieSerializer(movie, data=request.data)
+        if serializer.is_valid():
```
Comment (pavlopro) [inline cinema/views.py]: `serializer.is_valid(raise_exception=True)`

File: cinema/views.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    else:
```
Comment (pavlopro) [inline cinema/views.py]: Please explicitly check for "DELETE"

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+asgiref==3.8.1
+Django~=4.2.9
+django-rest-framework==0.1.0
+sqlparse~=0.5.0
+tzdata==2024.2
+
```
Comment (pavlopro) [inline requirements.txt]: remove blank line

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.idea
+db.sqlite3
+.db3
+.pyc
+venv/
```
Comment (pavlopro) [inline .gitignore]: add blank line

**Review by pavlopro — CHANGES_REQUESTED** [review]
Remove pycache from PR

File: cinema/views.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+@api_view(["GET", "PUT", "DELETE"])
+def movie_detail(request, pk):
+    movie = get_object_or_404(Movie, pk=pk)
+    if request.method == 'GET':
+        serializer = MovieSerializer(movie)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    elif request.method == 'PUT':
+        serializer = MovieSerializer(movie, data=request.data)
+        if serializer.is_valid(raise_exception=True):
```
Comment (pavlopro) [inline cinema/views.py]: No need to have this `if` operator

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+jupyter~=1.0.0
+pip~=24.2
+attrs~=23.2.0
+colorama~=0.4.6
+PyYAML~=6.0.1
+Jinja2~=3.1.3
+requests~=2.31.0
+djangorestframework~=3.15.2
```
Comment (pavlopro) [inline requirements.txt]: add blank line

**Review by pavlopro — CHANGES_REQUESTED** [review]
remove pycache from PR

**Review by pavlopro — CHANGES_REQUESTED** [review]
Remove pycache from PR

**Comment by TsvetanKichuk** [comment]
Hi, I can't remove it

File: cinema/migrations/__pycache__/0001_initial.cpython-312.pyc
Comment (vsmutok) [inline cinema/migrations/__pycache__/0001_initial.cpython-312.pyc:1]: Please remove  `__pycache__` from PR

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by pavlopro — CHANGES_REQUESTED** [review]
remove pycache from PR

**Pull request closed without merging**